Measure Length Error

Please see below GIF -- Measure Length Tool, in juxtaposition to Measure Area, is not capturing the measurement upon finalizing measurement parameters. 
![h2xmeasurelengtherror](https://user-images.githubusercontent.com/18521019/52887478-d8b8b980-3145-11e9-847d-1807aa98f513.gif)

I'm also noticing a warning when click either measure tool that should be investigated:
```
keybinding.js:165 warning: duplicate keybinding for "⌫-bubble"
keybinding.on @ keybinding.js:165
drawline @ draw_measure_line.js:183
(anonymous) @ call.js:4
context.install @ context.js:282
addLine.enter @ add_measure_line.js:42
context.enter @ context.js:253
subItems.append.attr.style.on.d @ tools.js:141
(anonymous) @ on.js:27
keybinding.js:165 warning: duplicate keybinding for "⌦-bubble"
keybinding.on @ keybinding.js:165
drawline @ draw_measure_line.js:184
(anonymous) @ call.js:4
context.install @ context.js:282
addLine.enter @ add_measure_line.js:42
context.enter @ context.js:253
subItems.append.attr.style.on.d @ tools.js:141
(anonymous) @ on.js:27
keybinding.js:165 warning: duplicate keybinding for "⎋-bubble"
keybinding.on @ keybinding.js:165
drawline @ draw_measure_line.js:185
(anonymous) @ call.js:4
context.install @ context.js:282
addLine.enter @ add_measure_line.js:42
context.enter @ context.js:253
subItems.append.attr.style.on.d @ tools.js:141
(anonymous) @ on.js:27
keybinding.js:165 warning: duplicate keybinding for "↩-bubble"
keybinding.on @ keybinding.js:165
drawline @ draw_measure_line.js:186
(anonymous) @ call.js:4
context.install @ context.js:282
addLine.enter @ add_measure_line.js:42
context.enter @ context.js:253
subItems.append.attr.style.on.d @ tools.js:141
(anonymous) @ on.js:27
```

--- a/modules/Hoot/managers/api.js
+++ b/modules/Hoot/managers/api.js
@@ -930,7 +930,7 @@ export default class API {
     /**
      * Delete a layer from the database
      *
-     * @param layerName - name of layer to delete
+     * @param layerId - map id of layer to delete
      * @returns {Promise<any>}
      */
     deleteLayer( layerId ) {

--- a/modules/behavior/draw_measure_area.js
+++ b/modules/behavior/draw_measure_area.js
@@ -5,12 +5,10 @@
  *******************************************************************************************************/
 
 import { utilRebind }           from '../util/rebind';
-import { utilKeybinding }       from '../util/keybinding';
 import { geoEuclideanDistance } from '../geo';
 
 export function behaviorDrawMeasureArea( context, svg ) {
     let dispatch        = d3.dispatch( 'move', 'click', 'undo', 'cancel', 'finish', 'dblclick' ),
-        keybinding      = utilKeybinding( 'measure' ),
         closeTolerance  = 4,
         tolerance       = 12,
         rectMargin      = 30,
@@ -135,12 +133,6 @@ export function behaviorDrawMeasureArea( context, svg ) {
                 ptArr.push( context.map().mouseCoordinates() );
             }
         }
-
-        svg
-            .append( 'g' )
-            .classed( `node point measure-vertex-${ nodeId }`, true )
-            .attr( 'transform', 'translate(' + c[ 0 ] + ',' + c[ 1 ] + ')' )
-            .attr( 'loc', mouseCoords );
 
         totDist     = totDist + segmentDist;
         segmentDist = 0;
@@ -341,18 +333,9 @@ export function behaviorDrawMeasureArea( context, svg ) {
         lengthLabel = label.append( 'tspan' ).text( '' );
         areaLabel   = label.append( 'tspan' ).text( '' );
 
-        keybinding
-            .on( '⌫', backspace )
-            .on( '⌦', del )
-            .on( '⎋', ret )
-            .on( '↩', ret );
-
         selection
             .on( 'mousedown.drawarea', mousedown )
             .on( 'mousemove.drawarea', mousemove );
-
-        d3.select( document )
-            .call( keybinding );
 
         return drawarea;
     }

--- a/modules/behavior/draw_measure_area.js
+++ b/modules/behavior/draw_measure_area.js
@@ -41,8 +41,8 @@ export function behaviorDrawMeasureArea( context, svg ) {
             pos     = point();
 
         element.on( 'dblclick', () => {
-            ret( element );
             polygon.classed( 'measure-complete', true );
+            ret( element );
         } );
 
         element.on( 'mousemove.drawarea', null );

--- a/modules/behavior/draw_measure_line.js
+++ b/modules/behavior/draw_measure_line.js
@@ -5,12 +5,10 @@
  *******************************************************************************************************/
 
 import { utilRebind }           from '../util/rebind';
-import { utilKeybinding }       from '../util/keybinding';
 import { geoEuclideanDistance } from '../geo';
 
 export function behaviorDrawMeasureLine( context, svg ) {
     let dispatch       = d3.dispatch( 'move', 'click', 'undo', 'cancel', 'finish', 'dblclick' ),
-        keybinding     = utilKeybinding( 'measure' ),
         closeTolerance = 4,
         tolerance      = 12,
         lastPoint      = null,
@@ -88,10 +86,6 @@ export function behaviorDrawMeasureLine( context, svg ) {
 
     function click() {
         let c = context.projection( context.map().mouseCoordinates() );
-
-        svg.append( 'g' )
-            .classed( 'node point measure-vertex-' + nodeId, true )
-            .attr( 'transform', 'translate(' + c[ 0 ] + ',' + c[ 1 ] + ')' );
 
         totDist     = totDist + segmentDist;
         segmentDist = 0;
@@ -179,18 +173,9 @@ export function behaviorDrawMeasureLine( context, svg ) {
     function drawline( selection ) {
         nodeId = 0;
 
-        keybinding
-            .on( '⌫', backspace )
-            .on( '⌦', del )
-            .on( '⎋', ret )
-            .on( '↩', ret );
-
         selection
             .on( 'mousedown.drawline', mousedown )
             .on( 'mousemove.drawline', mousemove );
-
-        d3.select( document )
-            .call( keybinding );
 
         return drawline;
     }

--- a/modules/modes/add_measure_area.js
+++ b/modules/modes/add_measure_area.js
@@ -9,8 +9,8 @@ import { modeBrowse }              from './browse';
 
 export function modeAddMeasureArea( context ) {
     let addArea = {
-        id: 'measure-add-line',
-        key: '6'
+        id: 'measure-add-area',
+        key: '7'
     };
 
     let svg = d3.select( '.data-layer.measure' ).select( 'svg' );

--- a/modules/modes/add_measure_area.js
+++ b/modules/modes/add_measure_area.js
@@ -22,12 +22,6 @@ export function modeAddMeasureArea( context ) {
     function finish( nodeId, ptArr ) {
         if ( ptArr.length < 3 ) {
             d3.select( '.data-layer.measure' ).selectAll( 'g' ).remove();
-        } else {
-            d3.selectAll( `.measure-line-${ nodeId }` ).each( function() {
-                d3.select( this.parentNode ).remove();
-            } );
-
-            d3.selectAll( `.measure-vertex-${ nodeId }` ).remove();
         }
 
         if ( d3.select( '.data-layer.measure' ).selectAll( 'g' ).size() ) {

--- a/modules/modes/add_measure_line.js
+++ b/modules/modes/add_measure_line.js
@@ -20,11 +20,14 @@ export function modeAddMeasureLine( context ) {
         .on( 'finish', finish );
 
     function finish( nodeId ) {
-        d3.selectAll( '.measure-line-' + nodeId ).each( function() {
-            d3.select( this.parentNode ).remove();
-        } );
+        d3.selectAll( '.measure-line-' + nodeId ).remove();
 
-        d3.selectAll( '.measure-vertex-' + nodeId ).remove();
+        d3.selectAll( '[class*="measure-line"]')
+            .filter(function() {
+                return d3.select(this).attr('loc1') === d3.select(this).attr('loc2');
+            }).each(function() {
+                this.parentNode.remove();
+            })
 
         if ( d3.select( '.data-layer.measure' ).selectAll( 'g' ).size() ) {
             d3.select( '.tools-toggle' ).text( 'Clear' );

--- a/modules/modes/add_measure_line.js
+++ b/modules/modes/add_measure_line.js
@@ -27,7 +27,7 @@ export function modeAddMeasureLine( context ) {
                 return d3.select(this).attr('loc1') === d3.select(this).attr('loc2');
             }).each(function() {
                 this.parentNode.remove();
-            })
+            });
 
         if ( d3.select( '.data-layer.measure' ).selectAll( 'g' ).size() ) {
             d3.select( '.tools-toggle' ).text( 'Clear' );

--- a/modules/ui/modes.js
+++ b/modules/ui/modes.js
@@ -7,7 +7,9 @@ import {
     modeAddLine,
     modeAddPoint,
     modeAddNote,
-    modeBrowse
+    modeBrowse,
+    modeAddMeasureLine,
+    modeAddMeasureArea
 } from '../modes';
 
 import { svgIcon } from '../svg';
@@ -21,6 +23,13 @@ export function uiModes(context) {
         modeAddArea(context),
         modeAddNote(context)
     ];
+
+    var toolModes = [
+        modeAddMeasureLine(context),
+        modeAddMeasureArea(context)
+    ].forEach(function(m){
+            context.keybinding().on(m.key, function() {context.enter(m);});
+    });
 
 
     function editable() {

--- a/modules/ui/tools.js
+++ b/modules/ui/tools.js
@@ -210,7 +210,6 @@ export function uiTools( context ) {
             } else {
                 toggle();
             }
-            toolsToggle.on( 'click', null );
         } );
 
         function toggle( cb ) {

